### DISCUSSION TITLE
Deprecating `dev` branch soon!

### DISCUSSION THREAD

**Discussion by jdangerx** [catalyst-cooperative/pudl#3180] [Announcements]
We are moving towards [trunk-based development](https://github.com/catalyst-cooperative/pudl/issues/3140), which means that we will be doing all of our development based on the `main` branch. This simplifies many development and automation workflows, which is good! However, it does mean some disruption to your workflows, sorry.

This will happen in the next couple of weeks - once the tooling (a first pass is in #3164, but I'm sure we'll have to work out a few kinks) works we'll have a firm timeline for deprecating the `dev` branch. 

* You will no longer be able to push to `dev`! Open PRs should be switched to the `main` branch instead.
* If you currently install PUDL from the `dev` branch, that will no longer get updates! Pin to one of the following options:
  * `main` will be the closest analog to `dev` going forwards - the bleeding edge.
  * `stable` will be the latest numbered version - we're hoping to get to a quarterly or monthly cadence here, so it should be fairly up-to-date.
  * `vXXXX.XX.XX` will be a specific numbered version - if you want to stay *really* stable.
* if you have links to our [documentation](https://catalystcoop-pudl.readthedocs.io/en/dev/) that point at `/en/dev`, they will no longer be updated - use `/en/latest` or `/en/stable` instead.

After deprecation, we will delete the branch after a few months and the things that are "not updated" will be "gone."

**Comment by zaneselvans** [comment]
Another place this change will show up is in the nightly build outputs that get pushed to S3 and GCS, currently with URLs like:

* `s3://pudl.catalyst.coop/dev/pudl.sqlite.gz`

Which will stop getting updated, eventually get removed, and be replaced with something like...
* `s3://pudl.catalyst.coop/nightly/pudl.sqlite.gz` (the last nightly build that succeeded; will be updated most mornings)
* `s3://pudl.catalyst.coop/stable/pudl.sqlite.gz` (the last versioned release; will update monthly to quarterly)
* `s3://pudl.catalyst.coop/v2023.12.01/pudl.sqlite.gz` (a specific versioned release, whose contents should not subsequently change)

**Comment by zaneselvans** [comment]
Hey y'all, through #3140, #3212, #3216 we have now deprecated the `dev` branch.  We now have 3 long-term branches:

- `main` is the default branch, which we branch off of and make PRs into.  Code there has been reviewed and passed CI, but may or may not result in successful nightly builds. This is bleeding edge and really just meant for develoment.
- `nightly` will always point at the commit associated with the last successful nightly builds. It's updated automatically and should never be pushed to. For now this will also be the default version for our documentation, Kaggle dataset, and Datasette deployment.
- `stable` will always point at the commit associated with the last versioned software & data release. It's also updated automatically and should never be pushed to.
